### ISSUE TITLE
False negative when using a variadic arg splat calling a function that accepts only one argument

### ISSUE THREAD

**Issue by ottumm** [microsoft/pyright#6353]
The following type-checks cleanly but errors at runtime. Desired behavior would be to report a type error, because calling `foo` with a splat will only work if the iterable being expanded only has one argument.

```python
def foo(others: tuple[int]) -> tuple[int]:
    ...


s = tuple(((n,) for n in range(10)))
foo(*s)
# at runtime: TypeError: foo() takes 1 positional argument but 10 were given
```

**VS Code extension or command-line**
Running Pyright from the command-line. Reproduced in latest version 1.1.334 and also reproduced going back to at least 1.1.320.

**Comment by erictraut** [comment]
The current behavior is correct. The type of `s` is `tuple[tuple[int], ...]`, so its length is not known to the type checker. If you change your code to produce a tuple of known length, pyright will generate an error if the length doesn't match the number of parameters supported by the callee.

Not surprisingly, mypy's behavior is the same as pyright's in the case.
